### PR TITLE
GEODE-4858: Update *DiskStore commands to use ResultModel and SingleGfshCommand

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jdeppe-pivotal** [comment]
precheckin is green

**Comment by pivotal-amurmann** [comment]
@jdeppe-pivotal Seems like this is bad now. Since it's so close to the finish line, do you want to fix it and get it merged?

**Comment by jdeppe-pivotal** [comment]
@jinmeiliao  @sboorlagadda @pdxrunner @PurelyApplied - if you have some time I'd appreciate a review so that this can finally be merged. Thanks!

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java [hunk 1]
```diff
@@ -101,41 +97,33 @@ public Result alterOfflineDiskStore(
...
-          result = ResultBuilder.createParsingErrorResult("Please provide a relevant parameter");
+          return ResultModel.createInfo("Please provide a relevant parameter");
         }
       }
       // Catch the IllegalArgumentException thrown by the modifyDiskStore function and sent the
     } catch (IllegalArgumentException e) {
-      String message = "Please check the parameters";
-      message += "\n" + e.getMessage();
-      result = ResultBuilder.createGemFireErrorResult(message);
+      return ResultModel.createInfo("Please check the parameters. " + e.getMessage());
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java]: There should be unit tests for the exceptions thrown by the command. If tests had previously existed, they would have shown that this is a change in behavior, i.e. from and error result to an info result.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java [hunk 2]
```diff
@@ -101,41 +97,33 @@ public Result alterOfflineDiskStore(
...
     } catch (IllegalStateException e) {
-      result = ResultBuilder.createGemFireErrorResult(e.getMessage());
+      return ResultModel.createError(e.getMessage());
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java]: Needs unit test.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java [hunk 3]
```diff
@@ -101,41 +97,33 @@ public Result alterOfflineDiskStore(
...
     } catch (CacheExistsException e) {
       // Indicates that the command is being used when a cache is open
-      result = ResultBuilder.createGemFireErrorResult("Cannot execute "
-          + CliStrings.ALTER_DISK_STORE + " when a cache exists (Offline command)");
+      return ResultModel.createInfo("Cannot execute " + CliStrings.ALTER_DISK_STORE
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterOfflineDiskStoreCommand.java]: Needs unit test.

File: geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java [hunk 4]
```diff
@@ -319,4 +329,79 @@ public void testCompactDiskStore() throws Exception {
...
+      r.put("A", "B");
+    });
+
+    // Should not be able to do this on a running system
+    String diskDirs = new File(server1.getWorkingDir(), DISKSTORE).getAbsolutePath();
+    gfsh.executeAndAssertThat(
+        String.format("upgrade offline-disk-store --name=%s --disk-dirs=%s", DISKSTORE, diskDirs))
+        .statusIsError().containsOutput("The disk is currently being used by another process");
+
+    server1.stop(false);
```
Comment (pdxrunner) [inline geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java]: Note: When you sync this PR with develop make sure the merges are all correct. When I merged develop to do this review, this line didn't merge correctly. I had to fix locally to build and run tests with the changes.

Comment (jdeppe-pivotal) [reply geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java]: I think think your develop might have been a bit behind.

Comment (pdxrunner) [reply geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java]: Could be. Thought I had updated in the checkout I was in for reviewing this, but maybe not.

**Pull request merged**